[CN-607][CN-606] Documentation for High Availability Mode feature (#68)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/high-availability-mode.adoc

= Running Highly Availiable Hazelcast clusters
:description: High Availability Mode guarantees that even if the Kubernetes node or whole availability zone is down (and all related Hazelcast members are terminated), you will not experience any data loss.

> Hazelcast offers the Partition Group Configuration and its types are ZONE_AWARE and NODE_AWARE, which forces Hazelcast members to store the given data partition backup inside a member located in a different availability zone or Kubernetes node. To learn about more, see the xref:hazelcast:clusters:partition-group-configuration.adoc[Platform documentation].

Reviewer: ```suggestion
In Hazelcast, you can enable and configure partition groups to force Hazelcast members to store backups of a data partition inside a member located in a different availability zone (ZONE_AWARE) or on a different Kubernetes node (NODE_AWARE). See xref:hazelcast:clusters:partition-group-configuration.adoc[Partition Group Configuration] for more details. In the Hazelcast Platform Operator, you can use these partition groups with High Availability Mode to guarantee that even if the Kubernetes node or whole availability zone is down, and all related Hazelcast members are terminated, you will not experience data loss.
```

> When using the ZONE_AWARE and NODE_AWARE partition grouping, a Hazelcast cluster spanning multiple availability zones and nodes should have an equal number of members in each availability zones and node. Otherwise, it results in uneven partition distribution among the members.

Reviewer: ```suggestion
== Using Multiple Availability Zones or Nodes

When using either type of partition grouping (ZONE_AWARE or NODE_AWARE) with a Hazelcast cluster that spans multiple availability zones and nodes, you must have an equal number of members in each zone or node. Otherwise, it results in the uneven distribution of partitions amongst the members.
```

Author: The wording seems ok to me but why do we need a separate section for that? Without multiple nodes/zones, this feature is meaningless so if we create a specific section like that it might create a misunderstanding among users that the operator supports one more topology.

Reviewer: Agreed.

> At the Platform Operator, users can configure type of partition-group and required Kubernetes scheduling policy(`topologySpreadConstraints`) to distribute members across availability zones and nodes automatically via modifying single `highAvailabilityMode` parameter.

Reviewer: ```suggestion
In the Hazelcast Platform Operator, you can configure the type of partition group and required Kubernetes scheduling policy(`topologySpreadConstraints`) to distribute members across availability zones and nodes automatically by modifying the `highAvailabilityMode` parameter.
```

> {description}

Reviewer: ```suggestion
```

== Configuring High Availability Mode 

> Below are the configuration options for the High Availability Mode feature.

Reviewer: In the other topics, we seem to have a standard line for the introduction of a new configuration item. Could we add something like this, if relevant? You can use the following configuration options in your XXXXX resource. For more detailed information, see the [API Reference](XXXXXXX).

Author: This feature has only one parameter so I think we do not need to redirect user API reference. WDYT?

Reviewer: Yes, that makes sense.

[cols="20%m,80%a"]
|===
|Field|Description

|`highAvailabilityMode`

> |The option to configure type partition-group and Kubernetes scheduling policy:

Reviewer: Configuration for partition groups and Kubernetes scheduling policy:
